### ISSUE TITLE
Uncaught TypeError: Cannot read property 'autocompleteIndex' of undefined

### ISSUE THREAD

**Issue by Quazz** [Smile-SA/smile-magento-elasticsearch#101]
When hovering over suggested results, every line returns this error in console log. 

`Uncaught TypeError: Cannot read property 'autocompleteIndex' of undefined`

The error seems to originate from scriptaculous controls.js

Clicking on a result results in a similar error and not the expected page load. 

I've searched for a while in hopes for a solution, but haven't found one yet, hopefully you can help me better with this problem.

**Comment by Quazz** [comment]
It seems that changing search.js line 77 

from             this.index = -1;
to              this.index = 1; (or any integer => 0)

Fixes all of these problems.

**Comment by Quazz** [comment]
After removing the offending sections in controls.js , everything works fine. They are only used for autocomplete, which this extension handles anyway (with very similar code)

**Comment by Quazz** [comment]
Removing  the offending sections (hover and click) from the autocomplete class of controls.js fixes this issue. 

Alternatively, if you're not using controls.js for anything, removing it entirely (or placing search.js higher in the hierarchy) should also work.

**Comment by Quazz** [comment]
Ok, this is real bizarre. If I use Chrome Developer console to disable cache on page reload, it works perfectly.

As soon as it's turned off, it goes back to throwing this error!

**Comment by Quazz** [comment]
If I disable Magento html output cache AND use Chrome Developer Console to disable local cache, it works perfectly every time. I have no idea how or why that is, though.

Update: And now sometimes it doesn't?!? Really don't know what to think about this anymore.

**Comment by Quazz** [comment]
This issue seems to not be present in 1.4 version.

**Comment by Quazz** [comment]
Update on this: 

It seems to be directly tied to how quickly results are returned from the search engine. Any significant delay, will cause this error to appear.

**Comment by Quazz** [comment]
After analyzing the XHR traffic, I noticed two requests per query.

This was causing all the issues. Turns out, there was a second search, using the normal template that was causing the issue. 

Removing the autocomplete segment from that template fixed the issue. (it's not necessary for it, and wasn't supposed to work regardless)

So, a lot of hairpulling and sighs for what turned out to be a fairly simple reason.
